core/state/statedb.go

```diff
@@ -201,6 +204,9 @@ func (s *StateDB) EnableWriteOnSharedStorage() {
 func (s *StateDB) StartPrefetcher(namespace string) {
 	s.prefetcherLock.Lock()
 	defer s.prefetcherLock.Unlock()
+	if s.noTrie {
```

Comment:
can we move it before  `s.prefetcherLock.Lock()`

Reply:
fixed